systemtest/src/main/java/io/strimzi/systemtest/interfaces/TestSeparator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.interfaces;
+
+import io.strimzi.test.timemeasuring.Operation;
+import io.strimzi.test.timemeasuring.TimeMeasuringSystem;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.util.Collections;
+import java.util.Map;
+
+@ExtendWith(ExtensionContextParameterResolver.class)
+public interface TestSeparator {
+    Logger LOGGER = LogManager.getLogger(TestSeparator.class);
+    String SEPARATOR_CHAR = "#";
+
+    static void printThreadDump() {
+        Map<Thread, StackTraceElement[]> allThreads = Thread.getAllStackTraces();
+        for (Thread thread : allThreads.keySet()) {
+            StringBuilder sb = new StringBuilder();
+            Thread key = thread;
+            StackTraceElement[] trace = allThreads.get(key);
+            sb.append(key).append("\r\n");
+            for (StackTraceElement aTrace : trace) {
+                sb.append(" ").append(aTrace).append("\r\n");
+            }
+            LOGGER.error(sb.toString());
+        }
+    }
+
+    @BeforeEach
+    default void beforeEachTest(TestInfo testInfo) {
+        TimeMeasuringSystem.setTestName(testInfo.getTestClass().get().getName(), testInfo.getTestMethod().get().getName());
+        TimeMeasuringSystem.startOperation(Operation.TEST_EXECUTION);
+        LOGGER.info(String.join("", Collections.nCopies(100, SEPARATOR_CHAR)));
+        LOGGER.info(String.format("%s.%s-STARTED", testInfo.getTestClass().get().getName(), testInfo.getTestMethod().get().getName()));
+    }
+
+    @AfterEach
+    default void afterEachTest(TestInfo testInfo, ExtensionContext context) {
+        TimeMeasuringSystem.stopOperation(Operation.TEST_EXECUTION);
+        if (context.getExecutionException().isPresent()) { // on failed
+            Throwable ex = context.getExecutionException().get();
+            if (ex instanceof OutOfMemoryError) {
+                LOGGER.error("Got OOM, dumping thread info");
+                printThreadDump();
```

Comment:
I re-use whole separator from enmasse team. The reason of this is to make sure, that application is not doing something nasty. Do you think it's not useful for us?

Reply:
TBH the most useful thing, if you're worried about OOME specifically, would be to run the JVM which is running the test with `-XX:+HeapDumpOnOutOfMemoryError` and `-XX:HeapDumpPath`. That will mean you get a heap dump (far more useful!). You could possibly add `-XX:OnOutOfMemoryError="kill -9 %p"` too. The problem with what you're doing is that the JVM as a whole could "run out of memory" and the first thread to hit this will get OOME, but that might not be the thread running the test. As the OOME propagates up the callstack a bunch of objects will become eligible for GC, so it's possible only that thread experiences OOME. But you missed it because it just wasn't your thread that happened to try to allocate when the JVM had no memory left. It's possible you finish the test successfully even though an OOME happened.

Reply:
I see, thanks for explanation. Going to remove this part.